pyccel/parser/semantic.py

```diff
@@ -2887,14 +2979,31 @@ def _visit_FunctionCall(self, expr):
                 return getattr(self, annotation_method)(expr)
 
         args = self._handle_function_args(expr.args)
+
         # Correct keyword names if scope is available
         # The scope is only available if the function body has been parsed
         # (i.e. not for headers or builtin functions)
-        if isinstance(func, FunctionDef) and func.scope:
+
+        if (isinstance(func, FunctionDef) and func.scope) or (isinstance(func, Interface) and func.is_annotated):
+            scope = func.scope if isinstance(func, FunctionDef) else func.syntactic_node.scope
```

Comment:
but here we only use to annotate the call arguments, so that case is not handled there,
also Interface does not have its own Scope, each function of the interface has its own Scope

Reply:
The case should be handled as in Fortran we can print function calls by explicitly naming arguments. For example the following code works in the `devel` branch:
```python
b = 2

def f(a : int = 3, b : int = 3):
    pass

f(b = 3)
```
In your branch it raises:
```
ERROR at Fortran compilation stage
Traceback (most recent call last):
  File "/home/emily/.local/bin/pyccel", line 8, in <module>
    sys.exit(pyccel())
  File "/home/emily/Code/pyccel/pyccel/commands/console.py", line 266, in pyccel
    execute_pyccel(filename,
  File "/home/emily/Code/pyccel/pyccel/codegen/pipeline.py", line 378, in execute_pyccel
    src_compiler.compile_module(compile_obj=mod_obj,
  File "/home/emily/Code/pyccel/pyccel/codegen/compiling/compilers.py", line 355, in compile_module
    self.run_command(cmd, verbose)
  File "/home/emily/Code/pyccel/pyccel/codegen/compiling/compilers.py", line 500, in run_command
    raise RuntimeError(err_msg)
RuntimeError: Failed to build module
/home/emily/Code/test/tmp/__pyccel__/tmp_conflict.f90:33:28:

   33 |       call f(a = 3_i64, b = 3_i64)
      |                            1
Error: Keyword argument ‘b’ at (1) is not in the procedure
```

Please add this case (or something similar demonstrating the problem) as a test

Reply:
Ah okay, I see, the scope is used only for keyword names, then it is fine to use any scope of an annotated function in the Interface

Reply:
fixed in [060c50e](https://github.com/pyccel/pyccel/pull/1728/commits/060c50ec49653c112df82346026d467713601bdb)